tests/framework/validators/AtLeastOneValidatorTest.php

```diff
@@ -0,0 +1,142 @@
+<?php
+namespace yiiunit\framework\validators;
+
+use yii\validators\AtLeastOneValidator;
+use yiiunit\data\validators\models\FakedValidationModel;
+use yiiunit\TestCase;
+
+/**
+ * @group validators
+ */
+class AtLeastOneValidatorTest extends TestCase
+{
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->mockApplication();
+    }
+
+    public function testValidateAttributeAndError()
+    {
+        $model = new FakedValidationModel;
+
+        $val = new AtLeastOneValidator(['alternativeAttributes' => 'attr2']);
+        $model->attr1 = null;
+        $model->attr2 = null;
+        $val->validateAttribute($model, 'attr1');
+        $this->assertCount(2, $model->getErrors(), 'Model should have 2 error attributes.');
+        $this->assertCount(1, $model->getErrors('attr1'));
+
+        $val = new AtLeastOneValidator(['alternativeAttributes' => 'attr2']);
+        $model->attr1 = '';
+        $model->attr2 = [];
+        $model->clearErrors();
+        $val->validateAttribute($model, 'attr1');
+        $this->assertCount(2, $model->getErrors(), 'Model should have 2 error attributes.');
+
+        $val = new AtLeastOneValidator([
+            'alternativeAttributes' => 'attr2',
+            'errorAttributes'=>'attr1',
+        ]);
+        $model->attr1 = null;
+        $model->attr2 = null;
+        $model->clearErrors();
+        $val->validateAttribute($model, 'attr1');
+        $this->assertCount(1, $model->getErrors(), 'Model should have 1 error attribute.');
```

Comment:
I'd suggest to use assertEquals on $model->getErrors() which makes the test easier to read compared to assertCount